### PR TITLE
feat wasm: expose functions

### PR DESCRIPTION
**Pull request by boddumanohar**
### Changes
- expose: `getAllocationFromAuthTicket`, `getReadPoolInfo`, `writePoolLock`, `redeemFreeStorage`, `decodeTicket`, `getBlobbersList`

### Fixes
- https://github.com/0chain/gosdk/issues/826
- https://github.com/0chain/gosdk/issues/825
- https://github.com/0chain/gosdk/issues/823
- https://github.com/0chain/gosdk/issues/828
- https://github.com/0chain/gosdk/issues/829
- https://github.com/0chain/gosdk/issues/827

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: wasmsdk/README.md [hunk 1]
```diff
@@ -183,6 +183,51 @@ updates allocation settings
...
+
+### zcn.sdk.redeemFreeStorage
+
+**Input**:
+> ticket string
+
+**Output**:
+> hash string
+
+### zcn.sdk.decodeTicket
```
Comment (cnlangzi) [inline wasmsdk/README.md]: it is better to rename it with `decodeAuthTicket` for consistent naming.

File: wasmsdk/README.md [hunk 2]
```diff
@@ -201,6 +246,13 @@ convert blobber urls to blobber ids
 **Output**:
 > []string
 
+### zcn.sdk.getBlobbersList
```
Comment (cnlangzi) [inline wasmsdk/README.md]: do you think it is better to name it with `getBlobbers`?

File: wasmsdk/README.md [hunk 3]
```diff
@@ -183,6 +183,51 @@ updates allocation settings
...
+### zcn.sdk.decodeTicket
+
+**Input**:
+> ticket string
+
+**Output**:
+> recipientPublicKey string, markerStr string, tokensInSAS uint64
+
+
+### zcn.sdk.writePoolLock
```
Comment (cnlangzi) [inline wasmsdk/README.md]: is it better to use `lockWritePool` or `lockTokensInWritePool` ?

File: wasmsdk/README.md [hunk 4]
```diff
@@ -183,6 +183,51 @@ updates allocation settings
...
+locks given number of tokes for given duration in read pool
```
Comment (cnlangzi) [inline wasmsdk/README.md]: `read pool` should be `write pool`

File: wasmsdk/allocation.go [hunk 5]
```diff
@@ -189,11 +196,88 @@ func getRemoteFileMap(allocationID string) ([]*fileResp, error) {
...
-		sdkLogger.Error("Failed to convert data to json format : %v", err)
 		return "", err
 	}
-	return string(b), nil
+
+	if recipientPublicKey != client.GetClientPublicKey() {
+		return "", fmt.Errorf("invalid_free_marker: free marker is not assigned to your wallet")
+	}
+
+	hash, _, err := sdk.CreateFreeAllocation(marker, lock)
```
Comment (cnlangzi) [inline wasmsdk/allocation.go]: it doesn't work anymore. please use 0box api to create free allocation.

File: wasmsdk/README.md [hunk 6]
```diff
@@ -183,6 +183,51 @@ updates allocation settings
 > hash: string
 
 
+### zcn.sdk.getAllocationFromAuthTicket
```
Comment (cnlangzi) [inline wasmsdk/README.md]: I would suggest to name it with `getAllocationWith(authTicket)`

**Review by cnlangzi — CHANGES_REQUESTED** [review]
some small fixes.

File: wasmsdk/README.md (hunk 1 above)
Comment (boddumanohar) [reply wasmsdk/README.md]: fixed in ee7cc18

File: wasmsdk/README.md (hunk 2 above)
Comment (boddumanohar) [reply wasmsdk/README.md]: fixed in ee7cc18

File: wasmsdk/README.md (hunk 3 above)
Comment (boddumanohar) [reply wasmsdk/README.md]: fixed in ee7cc18

File: wasmsdk/README.md (hunk 4 above)
Comment (boddumanohar) [reply wasmsdk/README.md]: fixed in ee7cc18

File: wasmsdk/README.md (hunk 6 above)
Comment (boddumanohar) [reply wasmsdk/README.md]: fixed in 9125454

File: wasmsdk/allocation.go (hunk 5 above)
Comment (boddumanohar) [reply wasmsdk/allocation.go]: removed freeallocation from wasm in 5657e95

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | expose-few-funcs  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | expose-few-funcs  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | expose-few-funcs  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | revert/978  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**